simtools/model/array_model.py

```diff
@@ -1,19 +1,485 @@
+                    version=self.modelVersion,
+                    label=self.label,
+                    modelFilesLocations=self._modelFilesLocations,
+                    filesLocation=self._filesLocation
+                )
+            else:
+                # Telescope name already exists.
+                # Finding the TelescopeModel and copying it.
+                for tel in self._telescopeModel:
+                    if tel.telescopeName != telModelName:
+                        continue
+                    self._logger.debug(
+                        'Copying tel model {} already loaded from DB'.format(tel.telescopeName)
+                    )
+                    telModel = copy(tel)
+                    break
+
+            self._telescopeModel.append(telModel)
+
+        # Checking whether the size of the telescope list and the layout match
+        if len(self._telescopeModel) != len(self.layout):
+            self._logger.warning(
+                'Number of telescopes in the list of telescope models does '
+                'not match the number of telescopes in the LayoutArray - something is wrong!'
+            )
+
+        # Changing parameters, if there are any in allParsToChange
+        if len(_allParsToChange) > 0:
+            for telData, telModel in zip(self.layout, self._telescopeModel):
+                if telData.name not in _allParsToChange.keys():
+                    continue
+                self._logger.debug('Changing {} pars of a {}: {}, ...'.format(
+                    len(_allParsToChange[telData.name]),
+                    telData.name,
+                    *_allParsToChange[telData.name]
+                ))
+                telModel.changeParameters(**_allParsToChange[telData.name])
+                telModel.setExtraLabel(telData.name)
+    # End of _buildArrayModel
+
+    def _getSingleTelescopeInfoFromArrayConfig(self, telName, telSize):
+        '''
+        arrayConfigData contains the default telescope models for each
+        telescope size and the list of specific telescopes.
+        For each case, the data can be given only as a name or
+        as a dict with 'name' and parameters to change.
+        This function has to identify these two cases and collect
+        the telescope name and the dict with the parameters to change.
+
+        Parameters
+        ----------
+        telName: str
+            Name of the telescope at the layout level (L-01, M-05, ...).
+        telSize: str
+            LST, MST or SST.
+        '''
+
+        def _proccessSingleTelescope(data):
+            '''
+            Parameters
+            ----------
+            data: dict or str
+                Piece of the arrayConfigData for one specific telescope.
+            '''
+            if isinstance(data, dict):
+                # Case 0: data is dict
+                if 'name' not in data.keys():
+                    msg = 'ArrayConfig has no name for a telescope'
```

Comment:
What does this error message mean? That the telName is not found in the list of telescope models given in ArrayConfig? Don't understand 'no name'